pkg/scheduler/plugins/gang/gang.go

```diff
@@ -157,8 +161,26 @@ func (gp *gangPlugin) OnSessionClose(ssn *framework.Session) {
 	for _, job := range ssn.Jobs {
 		if len(job.TaskStatusIndex[api.Pending]) != 0 {
 			glog.V(3).Infof("Gang: <%v/%v> allocated: %v, pending: %v", job.Namespace, job.Name, len(job.TaskStatusIndex[api.Allocated]), len(job.TaskStatusIndex[api.Pending]))
-			msg := fmt.Sprintf("%v/%v tasks in gang unschedulable: %v", len(job.TaskStatusIndex[api.Pending]), len(job.Tasks), job.FitError())
+			jobErrorMsg := job.FitError()
+			msg := fmt.Sprintf("%v/%v tasks in gang unschedulable: %v", len(job.TaskStatusIndex[api.Pending]), len(job.Tasks), jobErrorMsg)
 			ssn.Backoff(job, arbcorev1.UnschedulableEvent, msg)
+
+			// All pods Conditions need to be updated
+			for _, tasksMap := range job.TaskStatusIndex {
```

Comment:
How about add a function in `Session` to do that?

Reply:
Yeah. It will look cleaner. Let me make the change.

Reply:
@k82cn Did some clean up and move logic inside Session struct